### PR TITLE
Some things for time:*

### PR DESCRIPTION
**Pull request by Thog**
- Fully implement ISystemClock
- Implement ISteadyClock 1.0 cmds
- Add cmd 300 to IStaticService
- Add EphemeralNetwork to SystemClockType

### CODE REVIEW

File: Ryujinx.HLE/OsHle/Services/Time/IStaticService.cs [hunk 1]
```diff
@@ -17,7 +20,8 @@ public IStaticService()
                 { 1, GetStandardNetworkSystemClock },
                 { 2, GetStandardSteadyClock        },
                 { 3, GetTimeZoneService            },
-                { 4, GetStandardLocalSystemClock   }
+                { 4, GetStandardLocalSystemClock   },
+                { 300, CalculateMonotonicSystemClockBaseTimePoint }
```
Comment (Cyuubi) [inline Ryujinx.HLE/OsHle/Services/Time/IStaticService.cs]: Please align these.

File: Ryujinx.HLE/OsHle/Services/Time/IStaticService.cs [hunk 2]
```diff
@@ -56,5 +60,14 @@ public long GetStandardLocalSystemClock(ServiceCtx Context)
             return 0;
         }
 
+        public long CalculateMonotonicSystemClockBaseTimePoint(ServiceCtx Context)
+        {
+            long timeOffset = (long)(DateTime.UtcNow - StartupDate).TotalSeconds;
+            long systemClockContextEpoch = Context.RequestData.ReadInt64();
```
Comment (Cyuubi) [inline Ryujinx.HLE/OsHle/Services/Time/IStaticService.cs]: Please use CamelCase for these variables.

File: Ryujinx.HLE/OsHle/Services/Time/IStaticService.cs [hunk 3]
```diff
@@ -56,5 +60,14 @@ public long GetStandardLocalSystemClock(ServiceCtx Context)
...
+
+            Context.ResponseData.Write(timeOffset + systemClockContextEpoch);
```
Comment (Cyuubi) [inline Ryujinx.HLE/OsHle/Services/Time/IStaticService.cs]: Add a space below this.

File: Ryujinx.HLE/OsHle/Services/Time/ISteadyClock.cs [hunk 4]
```diff
@@ -9,12 +10,38 @@ class ISteadyClock : IpcService
...
+        private ulong TestOffset;
+
         public ISteadyClock()
         {
             m_Commands = new Dictionary<int, ServiceProcessRequest>()
             {
-                //...
+                {0, GetCurrentTimePoint },
+                {1, GetTestOffset },
+                {2, SetTestOffset }
```
Comment (Cyuubi) [inline Ryujinx.HLE/OsHle/Services/Time/ISteadyClock.cs]: Please align these.

File: Ryujinx.HLE/OsHle/Services/Time/ISteadyClock.cs [hunk 5]
```diff
@@ -9,12 +10,38 @@ class ISteadyClock : IpcService
...
             };
+            TestOffset = 0;
+        }
+
+        public long GetCurrentTimePoint(ServiceCtx Context)
+        {
+            Context.ResponseData.Write((long)(System.Diagnostics.Process.GetCurrentProcess().StartTime - DateTime.Now).TotalSeconds);
```
Comment (Cyuubi) [inline Ryujinx.HLE/OsHle/Services/Time/ISteadyClock.cs]: Add a space below this.

File: Ryujinx.HLE/OsHle/Services/Time/ISteadyClock.cs [hunk 6]
```diff
@@ -9,12 +10,38 @@ class ISteadyClock : IpcService
...
+            for (int i = 0; i < 0x10; i++)
+                Context.ResponseData.Write((byte)0);
+
+            return 0;
+        }
+
+        public long GetTestOffset(ServiceCtx Context)
+        {
+            Context.ResponseData.Write(TestOffset);
```
Comment (Cyuubi) [inline Ryujinx.HLE/OsHle/Services/Time/ISteadyClock.cs]: Please add a space below this.

File: Ryujinx.HLE/OsHle/Services/Time/ISteadyClock.cs [hunk 7]
```diff
@@ -9,12 +10,38 @@ class ISteadyClock : IpcService
...
+            return 0;
+        }
+
+        public long SetTestOffset(ServiceCtx Context)
+        {
+            this.TestOffset = Context.RequestData.ReadUInt64();
```
Comment (Cyuubi) [inline Ryujinx.HLE/OsHle/Services/Time/ISteadyClock.cs]: Please add a space below this.

**Comment by AcK77** [comment]
Thanks a lot!

**Pull request merged**